src/components/ListItem.jsx

```diff
@@ -1,5 +1,57 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+const currentTime = new Date();
+const numOfSecondsinMin = 60;
+const millisecondsToSeconds = 1000;
+
+export function ListItem({ name, item, listToken }) {
+	const { isChecked, id, totalPurchases, dateLastPurchased } = item;
+
+	if (isChecked) {
+		//check if it's been 24 hours since item was last marked as purchased
+		const timeDiffMinutes =
+			(currentTime.getTime() / millisecondsToSeconds -
+				dateLastPurchased.seconds) /
+			numOfSecondsinMin;
```

Comment:
Kudos on the math of it all. I always find this so daunting!

Reply:
Wow, @YolandaHaynes, thanks for the advice. That really cleans it up. It is helpful to know that when working with time, certain numbers are commonly accepted to use (rather than pulling them out as variables). I also like your suggestion for changing the variable name to `minutesSinceLastPurchased`! Thanks